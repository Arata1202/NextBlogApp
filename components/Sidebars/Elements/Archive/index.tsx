import { Listbox } from '@headlessui/react';
import { CalendarDaysIcon, ChevronUpDownIcon } from '@heroicons/react/24/solid';
import { archive } from '@/section/archive';
import { useState } from 'react';
import styles from '../../FixedSidebar/index.module.css';

export default function Archive() {
  const [selectedMonth, setSelectedMonth] = useState('');

  const handleArchiveChange = (value: string) => {
    if (value) {
      window.location.href = `/archive/${value}`;
      setSelectedMonth('');
    }
  };
  return (
    <div className="bg-white pt-8 px-4 border border-gray-300 py-5 mt-5">
      <h1 className={`${styles.profile} text-2xl text-center font-semibold flex justify-center`}>
        <CalendarDaysIcon className="h-8 w-8 mr-2" aria-hidden="true" />
        アーカイブ
      </h1>

      <Listbox value={selectedMonth} onChange={handleArchiveChange}>
        <div className="relative mt-5">
          <Listbox.Button
            style={{ height: '40px' }}
            className="relative w-full cursor-default rounded-md bg-white py-1.5 pl-3 pr-10 text-left text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-600 sm:text-sm sm:leading-6"
          >
            <span style={{ fontSize: '18px' }} className="block truncate">
              {selectedMonth
                ? `${selectedMonth.split('/')[0]}年${selectedMonth
                    .split('/')[1]
                    .replace(/^0+/, '')}月`
                : 'アーカイブを選択'}
            </span>
            <span className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
              <ChevronUpDownIcon className="h-7 w-7 text-gray-400" aria-hidden="true" />
            </span>
          </Listbox.Button>

          <Listbox.Options className="absolute z-10 mt-1 max-h-60 w-full overflow-auto rounded-md bg-white py-1 text-base shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none sm:text-sm">
            {archive.map((item, index) => (
              <Listbox.Option
                key={index}
                value={`${item.year}/${item.monthForPath}`}
                className={({ active }) =>
                  `relative cursor-default select-none py-2 pl-3 pr-9 text-gray-900 ${
                    active ? 'bg-[#eaf4fc]' : 'text-gray-900'
                  }`
                }
              >
                <span
                  className={`block truncate font-normal ${
                    selectedMonth === `${item.year}/${item.monthForPath}` ? 'font-semibold' : ''
                  }`}
                >
                  {`${item.year}年${item.month}月`}
                </span>
              </Listbox.Option>
            ))}
          </Listbox.Options>
        </div>
      </Listbox>
    </div>
  );
}
