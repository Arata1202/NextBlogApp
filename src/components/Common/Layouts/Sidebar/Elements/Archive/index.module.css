.ListBox {
  height: 40px;
}

.ListBoxSelect {
  font-size: 18px;
}
