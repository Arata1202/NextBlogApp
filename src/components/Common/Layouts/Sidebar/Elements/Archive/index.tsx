'use client';

import { useRouter } from 'next/navigation';
import { useState } from 'react';
import { useTheme } from 'next-themes';
import { Listbox } from '@headlessui/react';
import { CalendarDaysIcon, ChevronUpDownIcon } from '@heroicons/react/24/solid';
import styles from './index.module.css';
import { ARCHIVE_ARR } from '@/constants/archive';

export default function Archive() {
  const { theme } = useTheme();
  const router = useRouter();

  const [selectedMonth, setSelectedMonth] = useState('');

  const handleArchiveChange = (value: string) => {
    setSelectedMonth(value);
    router.push(`/archive/${value}`);
  };

  return (
    <>
      <div
        className={`pt-8 px-4 border py-5 mt-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
      >
        <div className={`text-2xl text-center font-semibold flex justify-center`}>
          <CalendarDaysIcon className="h-8 w-8 mr-2" />
          アーカイブ
        </div>

        <Listbox value={selectedMonth} onChange={handleArchiveChange}>
          <div className="relative mt-5">
            <Listbox.Button
              className={`${styles.ListBox} relative w-full cursor-pointer rounded-md py-1.5 pl-3 pr-10 text-left shadow-sm ring-1 ring-inset focus:ring-2 focus:ring-blue-500 focus:outline-none sm:text-sm sm:leading-6 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
            >
              <span className={`${styles.ListBoxSelect} block truncate text-gray-500`}>
                {selectedMonth
                  ? `${selectedMonth.split('/')[0]}年${selectedMonth
                      .split('/')[1]
                      .replace(/^0+/, '')}月`
                  : 'アーカイブを選択'}
              </span>
              <span className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
                <ChevronUpDownIcon
                  className={`h-7 w-7 ${theme === 'dark' ? 'text-gray-500' : 'text-gray-300'}`}
                />
              </span>
            </Listbox.Button>

            <Listbox.Options
              className={`absolute z-10 mt-1 max-h-60 w-full overflow-auto rounded-md py-1 text-base shadow-lg ring-1 ring-opacity-5 focus:outline-none sm:text-sm ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
            >
              {ARCHIVE_ARR.map((item, index) => (
                <Listbox.Option
                  key={index}
                  value={`${item.year}/${item.month.padStart(2, '0')}`}
                  className={`relative cursor-pointer select-none py-2 pl-3 pr-9 hover:text-blue-500 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
                >
                  {`${item.year}年${item.month}月`}
                </Listbox.Option>
              ))}
            </Listbox.Options>
          </div>
        </Listbox>
      </div>
    </>
  );
}
