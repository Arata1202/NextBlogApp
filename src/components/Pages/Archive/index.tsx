import { Article, Tag } from '@/types/microcms';
import AdUnit from '@/components/ThirdParties/GoogleAdSense/Elements/AdUnit';
import Pagination from '@/components/Common/Pagination';
import ArticleList from '@/components/Common/ArticleList';
import PageHeading from '@/components/Common/PageHeading';

type Props = {
  year: string;
  month: string;
  articles: Article[];
  totalCount: number;
  current?: number;
  recentArticles?: Article[];
  tags: Tag[];
};

export default function ArchivePage({
  year,
  month,
  articles,
  totalCount,
  current,
  recentArticles,
  tags,
}: Props) {
  return (
    <>
      <PageHeading year={year} month={month} />
      <ArticleList articles={articles} recentArticles={recentArticles} tags={tags} />
      <Pagination
        totalCount={totalCount}
        current={current}
        basePath={`/archive/${year}/${month}`}
      />
      <AdUnit slot="5969933704" style={{ marginTop: '1.25rem' }} />
    </>
  );
}
