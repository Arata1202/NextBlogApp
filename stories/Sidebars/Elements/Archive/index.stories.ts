import type { Meta, StoryObj } from '@storybook/react';
import Archive from '@/components/Sidebars/Elements/Archive';

const meta: Meta<typeof Archive> = {
  title: 'Sidebars/Elements/Archive',
  component: Archive,
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    isDarkMode: { control: 'boolean' },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    isDarkMode: false,
  },
};
