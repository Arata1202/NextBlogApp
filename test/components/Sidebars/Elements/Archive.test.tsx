import { render } from '@testing-library/react';
import Archive from '@/components/Sidebars/Elements/Archive';

describe('Archive', () => {
  test('スナップショット（Archive）', () => {
    const { asFragment } = render(<Archive />);
    expect(asFragment()).toMatchSnapshot();
  });
});
