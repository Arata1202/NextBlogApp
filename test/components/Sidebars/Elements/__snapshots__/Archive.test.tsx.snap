// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive スナップショット（Archive） 1`] = `
<DocumentFragment>
  <div
    class="bg-white pt-8 px-4 border border-gray-300 py-5 mt-5"
  >
    <h1
      class="profile text-2xl text-center font-semibold flex justify-center"
    >
      <svg
        aria-hidden="true"
        class="h-8 w-8 mr-2"
        data-slot="icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.75 12.75a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM7.5 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM8.25 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM9.75 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM10.5 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM12 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM12.75 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM14.25 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM15 17.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM16.5 15.75a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5ZM15 12.75a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0ZM16.5 13.5a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
        />
        <path
          clip-rule="evenodd"
          d="M6.75 2.25A.75.75 0 0 1 7.5 3v1.5h9V3A.75.75 0 0 1 18 3v1.5h.75a3 3 0 0 1 3 3v11.25a3 3 0 0 1-3 3H5.25a3 3 0 0 1-3-3V7.5a3 3 0 0 1 3-3H6V3a.75.75 0 0 1 .75-.75Zm13.5 9a1.5 1.5 0 0 0-1.5-1.5H5.25a1.5 1.5 0 0 0-1.5 1.5v7.5a1.5 1.5 0 0 0 1.5 1.5h13.5a1.5 1.5 0 0 0 1.5-1.5v-7.5Z"
          fill-rule="evenodd"
        />
      </svg>
      アーカイブ
    </h1>
    <div
      class="relative mt-5"
      data-headlessui-state=""
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        class="relative w-full cursor-default rounded-md bg-white py-1.5 pl-3 pr-10 text-left text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-600 sm:text-sm sm:leading-6"
        data-headlessui-state=""
        id="headlessui-listbox-button-:r2:"
        style="height: 40px;"
        type="button"
      >
        <span
          class="block truncate"
          style="font-size: 18px;"
        >
          アーカイブを選択
        </span>
        <span
          class="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2"
        >
          <svg
            aria-hidden="true"
            class="h-7 w-7 text-gray-400"
            data-slot="icon"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.47 4.72a.75.75 0 0 1 1.06 0l3.75 3.75a.75.75 0 0 1-1.06 1.06L12 6.31 8.78 9.53a.75.75 0 0 1-1.06-1.06l3.75-3.75Zm-3.75 9.75a.75.75 0 0 1 1.06 0L12 17.69l3.22-3.22a.75.75 0 1 1 1.06 1.06l-3.75 3.75a.75.75 0 0 1-1.06 0l-3.75-3.75a.75.75 0 0 1 0-1.06Z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
